bee-api/bee-rest-api/src/endpoints/mod.rs

```diff
@@ -16,40 +16,82 @@ use storage::StorageBackend;
 
 use crate::types::body::{DefaultErrorResponse, ErrorBody};
 
-use bee_gossip::NetworkCommandSender;
-use bee_ledger::workers::consensus::ConsensusWorker;
+use bee_gossip::{Keypair, NetworkCommandSender, PeerId};
+use bee_ledger::workers::consensus::{ConsensusWorker, ConsensusWorkerCommand};
 use bee_protocol::workers::{
-    config::ProtocolConfig, MessageRequesterWorker, MessageSubmitterWorker, PeerManager, PeerManagerResWorker,
-    RequestedMessages,
+    config::ProtocolConfig, MessageRequesterWorker, MessageSubmitterWorker, MessageSubmitterWorkerEvent, PeerManager,
+    PeerManagerResWorker, RequestedMessages,
 };
 use bee_runtime::{
-    node::{Node, NodeBuilder},
+    event::Bus,
+    node::{Node, NodeBuilder, NodeInfo},
+    resource::ResourceHandle,
     worker::{Error as WorkerError, Worker},
 };
 use bee_tangle::{Tangle, TangleWorker};
 
 use async_trait::async_trait;
 use log::{error, info};
+use tokio::sync::mpsc;
 use warp::{http::StatusCode, Filter, Rejection, Reply};
 
-use std::{any::TypeId, convert::Infallible};
+use std::{any::TypeId, convert::Infallible, sync::Arc};
 
 pub(crate) type NetworkId = (String, u64);
 pub(crate) type Bech32Hrp = String;
+#[cfg(feature = "dashboard")]
+pub(crate) type DashboardUsername = String;
 
 pub(crate) const CONFIRMED_THRESHOLD: u32 = 5;
 
-pub async fn init_full_node<N: Node>(
-    rest_api_config: RestApiConfig,
-    protocol_config: ProtocolConfig,
-    network_id: NetworkId,
-    bech32_hrp: Bech32Hrp,
-    node_builder: N::Builder,
-) -> N::Builder
+pub async fn init_full_node<N: Node>(init_config: InitConfigFullNode, node_builder: N::Builder) -> N::Builder
 where
     N::Backend: StorageBackend,
 {
-    node_builder.with_worker_cfg::<ApiWorkerFullNode>((rest_api_config, protocol_config, network_id, bech32_hrp))
+    node_builder.with_worker_cfg::<ApiWorkerFullNode>(init_config)
+}
+
+pub async fn init_entry_node<N: Node>(init_config: InitConfigEntryNode, node_builder: N::Builder) -> N::Builder
+where
+    N::Backend: StorageBackend,
+{
+    node_builder.with_worker_cfg::<ApiWorkerEntryNode>(init_config)
+}
+
+pub struct InitConfigFullNode {
+    pub node_id: PeerId,
+    pub node_keypair: Keypair,
+    pub rest_api_config: RestApiConfig,
+    pub protocol_config: ProtocolConfig,
+    pub network_id: NetworkId,
+    pub bech32_hrp: Bech32Hrp,
+    #[cfg(feature = "dashboard")]
+    pub dashboard_username: DashboardUsername,
+}
+
+pub struct InitConfigEntryNode {
+    pub rest_api_config: RestApiConfig,
+}
+
+pub struct ApiArgsFullNode<B: StorageBackend> {
+    pub node_id: PeerId,
+    pub node_keypair: Keypair,
+    pub rest_api_config: RestApiConfig,
+    pub protocol_config: ProtocolConfig,
+    pub network_id: NetworkId,
+    pub bech32_hrp: Bech32Hrp,
+    #[cfg(feature = "dashboard")]
+    pub dashboard_username: DashboardUsername,
```

Comment:
This is really nitpicky, but since this is reliant on a feature flag I would usually place this at the end of the list of fields, so there's some separation.

Reply:
Good idea done :d